### PR TITLE
Add: List of hex-codes

### PR DESCRIPTION
**Pull request by PROxZIMA**
Creates an array of 0..16777216, maps them to hex string and adds
padding to them

### CODE REVIEW

**Comment by amos-kibet** [comment]
Thanks, I'll have a look once I'm on my laptop and merge it.

File: app.js [hunk 1]
```diff
@@ -1,5 +1,6 @@
-// To Do: Include all 256 hex color codes.
-const colors = ["#AA869E", "#36271D", "#30E7C8", "#9DA93C", "#D43494", "#62B5FF", "#EFBBC6", "#522E48", "#542BD5", "#009AE0", "#532671", "#791926", "#173D21", "#BB75D9", "#AB3AB9", "#7D51F5"];
+// Include all 256*256*256 hex color codes.
+const colors = [...Array(16777216).keys()].map(i => `#${i.toString(16).padStart(6, '0')}`);
+console.log('hi');
```
Comment (amos-kibet) [inline app.js]: Delete this console message

**Review by amos-kibet — CHANGES_REQUESTED** [review]
All look good. You're smart, I like how you have used padStart method. I had never heard of it before

File: app.js (hunk 1 above)
Comment (PROxZIMA) [reply app.js]: Done!

**Comment by PROxZIMA** [comment]
Haha! It just converts `256*256*256` integers to `hex` and then left pad them with `0`

**Comment by amos-kibet** [comment]
What I read is that with padStart(x, y), the x parameter specifies the length of the resulting value after 'trimming', while the y parameter replaces the 'trimmed' values. So, I was expecting all your 'trimmed' values will be replaced with 0, then they were not. Maybe you can explain a little how that method works.

Otherwise, your solution is great. Your programmatic approach is incredible.

**Comment by PROxZIMA** [comment]
Not exactly trimming. According to [MDN](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/String/padStart)
> The padStart() method pads the current string with another string (multiple times, if needed) until the resulting string reaches the given length. The padding is applied from the start of the current string.

So if `i` is `15`, it's hex string will be `f` and it will be padded with `0` until it's length becomes `6`, resulting `00000f`. That's all

**Comment by amos-kibet** [comment]
I think I will read and practice  more on it to understand it better.
It was nice interacting with you sir. If I may ask, could we collaborate more? I'm self-teaching coding and preparing to transition from electrical field to software. I could use your skills as we collaborate. If this sounds great to you, we could talk about objectives, projects we could build together, and learnings we could take.

**Comment by PROxZIMA** [comment]
Hey don't call me sir!! I'm still an under graduate.
> If I may ask, could we collaborate more?

Sure I would love to work with you. As I'm still learning stuffs using them in an actual project will be beneficial

**Comment by amos-kibet** [comment]
Sir is a polite and respectful way of addressing a man, but that is noted. Do you have projects we can collaborate on building?

**Comment by PROxZIMA** [comment]
I'm not actually learning web development but it's my hobby so I have one
https://github.com/PROxZIMA/prism
This a simple start-page I forked from [Prismatic Night](https://github.com/3r3bu5x9/Prismatic-Night). Still have tons of work remaining.

If you have anything from your side then I'm glad to hear!!

**Pull request merged**